app/javascript/components/GooglePayButton.js

```diff
@@ -0,0 +1,150 @@
+// Google Pay not restricted to Android devices.
+const isAndroid = navigator.userAgent.toLowerCase().indexOf('android') >= 0;
+
+export class GooglePayButton extends Component {
+  props: Props;
+  state: State;
+  buttonRef: any;
+  constructor(props: Props) {
+    super(props);
+    // $FlowIgnore
+    this.buttonRef = React.createRef();
+    this.state = {
+      paymentsClient: undefined,
+      googlePaymentInstance: undefined,
+      ready: false,
+    };
+  }
+
+  createGooglePayButton = () => {
+    $.ajax({
+      url: 'https://pay.google.com/gp/p/js/pay.js',
+      dataType: 'script',
+    }).then(() => {
+      if (window.google) {
+        const paymentsClient = new window.google.payments.api.PaymentsClient({
+          environment:
+            process.env.NODE_ENV === 'production' ? 'PRODUCTION' : 'TEST',
+        });
+
+        const button = paymentsClient.createButton({
+          onClick: this.onClick,
+          buttonType: 'short',
+        });
+        this.buttonRef.current.appendChild(button);
+
+        this.setState({ paymentsClient });
+      }
+    });
+  };
+
+  setupGooglePay() {
+    const { client } = this.props;
+    const { paymentsClient } = this.state;
+    if (client && paymentsClient) {
+      googlePayment.create({ client }, (err, googlePaymentInstance) => {
+        paymentsClient
+          .isReadyToPay({
+            allowedPaymentMethods: googlePaymentInstance.createPaymentDataRequest()
+              .allowedPaymentMethods,
+          })
+          .then(response => {
+            if (response.result) {
+              this.setState({
+                paymentsClient,
+                googlePaymentInstance,
+                ready: true,
+              });
+            }
+          });
+      });
+    }
+  }
+
+  componentDidMount() {
+    this.createGooglePayButton();
+    this.setupGooglePay();
+  }
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (
+      prevProps.client !== this.props.client ||
+      prevState.paymentsClient !== this.state.paymentsClient
+    ) {
+      this.setupGooglePay();
+    }
+  }
+
+  onClick = (e: any) => {
+    e.preventDefault();
+    const { ready, googlePaymentInstance, paymentsClient } = this.state;
+    if (ready && googlePaymentInstance && paymentsClient) {
+      const paymentDataRequest = googlePaymentInstance.createPaymentDataRequest(
+        {
+          merchantId: undefined,
+          transactionInfo: {
+            currencyCode: this.props.currency,
+            totalPrice: this.props.amount.toString(),
+            totalPriceStatus: 'FINAL',
+          },
+          cardRequirements: {
+            billingAddressRequired: true,
+          },
+        }
+      );
+
+      paymentsClient
+        .loadPaymentData(paymentDataRequest)
+        .then(paymentData => {
+          googlePaymentInstance.parseResponse(paymentData, (err, result) => {
+            if (err) {
+              // TODO
```

Comment:
will error handling be a pat of this PR or did you intentionally leave this for later?

Reply:
Intentionally left out in this PR since at the moment we're only demoing in a sandbox (which is also why i've disabled the feature behind the feature flag until it's production ready).